### ISSUE TITLE
Scope efforts for WASM compat 

### ISSUE THREAD

**Issue by gbotrel** [ConsenSys/gnark#716]
By the way, currently we are blocked by two things:

1. Hints are still using the reflect mechanism, named hints are already implemented but not used. Maybe tinygo 0.28 might solve that, but it is yet not sure. 
2. Cbor is now serializing an object that cannot be easy replaced by Gob or other tools, that breaks our attempts to test the last develop gnark code.

_Originally posted by @p4u in https://github.com/ConsenSys/gnark/issues/74#issuecomment-1570590007_

**Comment by ivokub** [comment]
Duplicate of #74
